File: backend/src/main/java/com/Fijalkowskim/SpaceFarmManagmentSystem/requestmodels/StageRequest.java
package com.Fijalkowskim.SpaceFarmManagmentSystem.requestmodels;

import com.Fijalkowskim.SpaceFarmManagmentSystem.models.Control;
import com.Fijalkowskim.SpaceFarmManagmentSystem.models.dictionaries.StageType;
import lombok.Data;

import java.util.Date;
import java.util.Set;

@Data
public class StageRequest {
    private StageType stageType;

    private Date startStageDate;

    private Date finishStageDate;

    private String comment;

    private Set<Control> controls;

    private long cultivationId;
}

File: frontend/src/models/requestmodels/StageRequest.js
export class StageRequest {
  constructor(
    stageType,
    startStageDate = new Date(),
    finishStageDate,
    comment = "",
    controls,
    cultivationId
  ) {
    this.stageType = stageType;
    this.startStageDate = startStageDate;
    this.finishStageDate = finishStageDate;
    this.comment = comment;
    this.controls = controls;
    this.cultivationId = cultivationId;
  }
}
